adapters/Vungle/VungleAdapter/GADMAdapterVungleRouter.m

```diff
@@ -192,15 +209,11 @@ - (void)removeDelegate:(nonnull id<GADMAdapterVungleDelegate>)delegate {
         GADMAdapterVungleMapTableRemoveObjectForKey(_delegates, delegate.desiredPlacement);
       }
     }
-  }
-}
-
-- (void)removeBannerDelegate:(nonnull id<GADMAdapterVungleDelegate>)delegate {
-  if ([delegate respondsToSelector:@selector(isBannerAd)] && [delegate isBannerAd]) {
+  } else if ([delegate respondsToSelector:@selector(isBannerAd)] && [delegate isBannerAd]) {
     @synchronized(_bannerDelegates) {
       if (delegate && [_bannerDelegates objectForKey:delegate.bannerRequest]) {
         NSMutableOrderedSet *set = [[_bannerDelegates objectForKey:delegate.bannerRequest] mutableCopy];
-        if ([set containsObject:delegate]) {
+        if ([set containsObject:delegate] && (delegate.bannerState != BannerRouterDelegateStatePlaying) && (delegate.bannerState != BannerRouterDelegateStateClosing)) {
```

Comment:
AdMob SDK always releases the Banner delegate early, so if the Banner delegate is in `BannerRouterDelegateStatePlaying` or  `BannerRouterDelegateStateClosing` state,  the adapter  will not remove this delegate to ensure receiving subsequent callback methods `-(void)willCloseAd:didDownload:` and `-(void)didCloseAd:didDownload:`. And after calling the callback method `-(void)didCloseAd:didDownload:`, the Banner delegate state will become `BannerRouterDelegateStateClosed` which could be removed from  `_bannerDelegates`. Thanks a lot. 🙂